Loop/Managers/WatchDataManager.swift

```diff
@@ -350,21 +403,28 @@ extension WCSession {
             "* outstandingUserInfoTransfers: \(outstandingUserInfoTransfers)",
             "* receivedApplicationContext: \(receivedApplicationContext)",
             "* remainingComplicationUserInfoTransfers: \(remainingComplicationUserInfoTransfers)",
-            "* complicationUserInfoTransferInterval: \(round(complicationUserInfoTransferInterval.minutes)) min",
             "* watchDirectoryURL: \(watchDirectoryURL?.absoluteString ?? "nil")",
         ].joined(separator: "\n")
     }
-
-    fileprivate var complicationUserInfoTransferInterval: TimeInterval {
+    
+    fileprivate func complicationUserInfoTransferInterval(bedtime: Date?) -> TimeInterval {
         let now = Date()
-        let timeUntilMidnight: TimeInterval
+        let timeUntilRefresh: TimeInterval
 
         if let midnight = Calendar.current.nextDate(after: now, matching: DateComponents(hour: 0), matchingPolicy: .nextTime) {
-            timeUntilMidnight = midnight.timeIntervalSince(now)
+            // we can have a more frequent refresh rate if we only refresh when it's likely the user is awake (based on HealthKit sleep data)
+            if let nextBedtime = bedtime {
+                let timeUntilBedtime = nextBedtime.timeIntervalSince(now)
+                timeUntilRefresh = timeUntilBedtime > TimeInterval(0) ? timeUntilBedtime : midnight.timeIntervalSince(now)
```

Comment:
We might defensively make the condition in this ternary also check an upper bound on `timeUntilBedtime`, i.e. `(0..<TimeInterval(hours: 24)).contains(timeUntilBedtime)`, to ensure that complication transfers aren't slowed down if `bedtime` somehow ended up being a Date further in the future.